pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ConnectionPoolTest.java

```diff
@@ -138,15 +146,19 @@ public void testDoubleIpAddress() throws Exception {
         client.close();
 
         eventLoop.shutdownGracefully();
+        executorProvider.shutdownNow();
```

Comment:
Instead of this, you could have `@Cleanup("shutdownNow")` above the `ScheduledExecutorProvider executorProvider = ...` line.